LICENSE.AGPL

```diff
@@ -0,0 +1,661 @@
+                    GNU AFFERO GENERAL PUBLIC LICENSE
```

Comment:
Hi Ismail,

Many open-source software organizations ask contributors to sign their copyrights to the primary maintaining organization, for multiple reasons - this is fairly standard practice.  For example, even the Free Software Foundation does it:

https://www.gnu.org/licenses/why-assign

I have personally signed FSF copyright-release forms for the (fairly small) contributions I've made to GNU software in the (somewhat distant) past.  

One reason for this (probably the main reason the FSF does it) is to have better legal leverage in the event of litigation: e.g., if some company tries to "steal" the free software and sell some variation of it as proprietary software without releasing the source or otherwise complying with the terms of the open-source license.  

Another common reason is that the company holding the copyright can "sell exceptions" to the terms of the GPL; this is the traditional (fairly successful) funding model that Qt/Trolltech used for example before it was bought by Nokia and Nokia imploded.  The FSF talks about this, too - they don't like it all that much (understandably given the fairly purist positions they usually take), but they call this practice "acceptable, though only barely". :)

https://www.fsf.org/blogs/rms/assigning-copyright

Regardless, all this only applies to files and modules that were primarily written in DEDIS (and may incorporate small contributions from non-DEDIS/EPFL people); there is no reason this needs to prevent us from incorporating and building on open-source code copyrighted by other organizations and released with compatible open-source licenses, nor does this prevent other organizations from taking DEDIS code and building on it, modifying it, and using it more-or-less in whatever ways they want.  That's what the open-source license is designed to allow.  Note that an explicit exception is made to the contributor release rule for code that is explicitly documented to be copyrighted by someone else, so this is all just about the "default" copyright that applies to the code that was written mostly by DEDIS and for which DEDIS/EPFL is the primary copyright holder.

Does that help?